### ISSUE TITLE
Evaluate 'default' lazily in 'dict$get()'

### ISSUE THREAD

**Issue by tdeenes** [randy3k/collections#12]
Maybe some users would find the following sugar useful:
```r
longRunningFn <- function() {
  message("Evaluating started...")
  Sys.sleep(5)
  FALSE
}
d <- collections::dict()$set("existing_key", TRUE)
## instead of this:
value <- 
  if (d$has("existing_key")) {
    d$get("existing_key")
  } else {
    longRunningFn()
  }
## we could write:
value <- d$get("existing_key", default = longRunningFn())
```

Currently the `default` argument is evaluated even if the key exists in the dictionary.

**Comment by randy3k** [comment]
closed by 69a394ffe38299234a0e087aa0b237c31617ecc8

**Comment by tdeenes** [comment]
Thanks!
